internal/globalconfig/globalconfig_test.go

```diff
@@ -18,3 +21,39 @@ func TestHeaderTag(t *testing.T) {
 	assert.Equal(t, "tag1", cfg.headersAsTags.Get("header1"))
 	assert.Equal(t, "tag2", cfg.headersAsTags.Get("header2"))
 }
+
+func TestSetStatsCarrier(t *testing.T) {
+	sc := internal.NewStatsCarrier(&statsd.NoOpClient{})
+	SetStatsCarrier(sc)
+	assert.NotNil(t, cfg.statsCarrier)
+	//reset globalconfig for other tests
+	cfg.statsCarrier = nil
+}
+
+func TestPushStat(t *testing.T) {
+	var tg statsdtest.TestStatsdClient
+	sc := internal.NewStatsCarrier(&tg)
+	sc.Start()
+	defer sc.Stop()
+	cfg.statsCarrier = sc
+	stat := internal.NewGauge("name", float64(1), nil, 1)
+	PushStat(stat)
+	calls := tg.CallNames()
+	assert.Len(t, calls, 1)
+	assert.Contains(t, calls, "name")
+	//reset globalconfig for other tests
+	cfg.statsCarrier = nil
```

Comment:
nit: I recommend just putting this in a defer earlier. e.g.

```
cfg.statsCarrier = sc
defer cfg.statsCarrier = nil // reset globalconfig for other tests
```

Reply:
I originally had them in a defer, but it caused the tests to flake - I assumed the issue was that`defer` might not run within the scope of a subtest - with the `defer`, I saw subtests ccessing the stats carrier before it had been cleared by another subtest, so they were incorrectly sharing state. Removing the defer worked.

Reply:
Hmmm yeah I suppose that's true for the subtests in `TestStatsCarrier`, but shouldn't be a problem here.

Actually, how about just using `t.Cleanup` instead? https://pkg.go.dev/testing#T.Cleanup